R/TealAppDriver.R

```diff
@@ -382,6 +382,17 @@ TealAppDriver <- R6::R6Class( # nolint: object_name.
       invisible(self)
     },
     #' @description
+    #' Extract `onlick` attribute from a `button` (found by `selector`).
+    #'
+    #' @param selector (`character(1)`) specifying the selector to be used to get the content of a `button`.
+    #'
+    #' @return The `character` vector.
+    get_onclick = function(selector) {
+      self$get_html_rvest(selector) %>%
+        rvest::html_nodes("button") %>%
+        rvest::html_attr("onclick")
+    },
```

Comment:
This is a very specific method for a single attribute, what do you think about converting it to a generic attribute getter? (that works for all HTML nodes besides `button`)

then replace all `app$get_onclick(selector)` with `app$get_attr(selector, "onclick"))`

```suggestion
    #' @description
    #' Extract `html` attribute from (found by a `selector`).
    #'
    #' @param selector (`character(1)`) specifying the selector to be used to get the content of a `button`.
    #'
    #' @return The `character` vector.
    get_attr = function(selector, attribute_name) {
      rvest::read_html(self$get_html("html")) %>%
        rvest::html_nodes(selector) %>%
        rvest::html_attr(attribute_name)
    }
```

Reply:
Note that `rvest::read_html` after piping from `app$get_html()` will add a `<body>` wrapper around the element.

Hence it might be better to use `rvest::read_html(self$get_html("html"))`

Ideally, we'd use `xml2::read_xml` after `app$get_html(selector)`, but that would add a new dependency :-\

Reply:
`rvest::read_html(self$get_html("html"))` is implemented as `self$get_html_rvest("html")`

Reply:
I like `get_attr` to be customized for a wider approach! Will include, and rewrite in few places where applicable.

Reply:
:facepalm: I've updated the suggestion to use `self$get_html_rvest("html")`

> I think we will replace `rvest` with `xml2` #1162 so no new dependencies. `rvest` under the hood is just a `xml2` all over the place : )

Cool!